lib/fsm.ex

```diff
@@ -3,7 +3,8 @@ defmodule Fsm do
     quote do
       import Fsm
 
-      defstruct [:state, :data]
+      defstruct [state: unquote(opts[:initial_state]),
+                 data:  unquote(opts[:initial_data])]
```

Comment:
This looks fine, but then I think we can avoid setting the same fields in line 13, and reduce it to `%__MODULE__{}`.